joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -221,6 +246,37 @@ JointTrajectoryController::on_configure(const rclcpp_lifecycle::State & previous
   // TODO(karsten1987): no lifecyle for subscriber yet
   // joint_command_subscriber_->on_activate();
 
+  // State publisher
+  {
+    int state_publish_rate = 50;
+    if (lifecycle_node_->has_parameter("state_publish_rate")) {
+      state_publish_rate = lifecycle_node_->get_parameter("state_publish_rate").get_value<int>();
+      RCLCPP_INFO_STREAM(
+        logger, "Controller state will be published at " <<
+          state_publish_rate << "Hz.");
+    }
+    state_publisher_period_ = rclcpp::Duration(1.0 / static_cast<double>(state_publish_rate));
```

Comment:
Can we fetch this as a double (who says Hz has to be integer?) to avoid that `static_cast` and let's add a non-zero positive check here, otherwise we may create a black hole

Reply:
So there's a problem with casting to doubles with parameters, I reported it.
https://github.com/ros2/rclcpp/issues/1009

As a workaround, I used integers.

Reply:
Caught up with the issue and fixed in [0212a21](https://github.com/ros-controls/ros2_controllers/pull/25/commits/0212a2104627ec58052144bc839496bfdc14d81a)